.github/workflows/release.yml

```diff
@@ -339,3 +339,68 @@ jobs:
               exit 1
             }
           }
+
+  onedrive:
+    name: OneDrive
+    runs-on: ubuntu-18.04
+    needs: preflight
+
+    steps:
+      - name: Check out Devolutions/actions
+        uses: actions/checkout@v3
+        with:
+          repository: Devolutions/actions
+          ref: master
+          token: ${{ secrets.DEVOLUTIONSBOT_TOKEN }}
+          path: ./.github/workflows
+
+      ## Devolutions Toolbox is required for OneDrive uploading
+
+      - name: Install Devolutions Toolbox
+        uses: ./.github/workflows/toolbox-install
+        with:
+          token: ${{ secrets.DEVOLUTIONSBOT_TOKEN }}
+
+      ## Download back the artifacts produced by the other jobs
+
+      - name: Download artifacts (action)
+        if: needs.preflight.outputs.dl-strategy == 'action'
+        uses: actions/download-artifact@v3
+        with:
+          name: devolutions-gateway
+
+      - name: Download artifacts (cli)
+        if: needs.preflight.outputs.dl-strategy == 'cli'
+        shell: pwsh
+        env:
+          GITHUB_TOKEN: ${{ secrets.DEVOLUTIONSBOT_TOKEN }}
+        run: gh run download ${{ needs.preflight.outputs.run }} -n devolutions-gateway --repo $Env:GITHUB_REPOSITORY
+
+      ## workflow_call: The same artifacts persist across the entire run, so the PowerShell/DevolutionsGateway directory will still exist from the CI workflow
+      - name: Manage artifacts
+        shell: pwsh
+        run: Remove-Item (Join-Path PowerShell DevolutionsGateway) -Recurse -ErrorAction Ignore
+
+      - name: Prepare upload
+        id: prepare
+        shell: pwsh
+        run: |
+          $destinationFolder = "${{ runner.temp }}/artifacts"
+          $version="${{ needs.preflight.outputs.version }}"
+
+          Write-Host "::set-output name=version::${version}.0"
+          Write-Host "::set-output name=files-to-upload::${destinationFolder}"
+
+          New-Item -Path "$destinationFolder" -ItemType "directory"
+          Move-Item -Path "./windows/x86_64/DevolutionsGateway-x86_64-${version}.msi" -Destination "$destinationFolder/DevolutionsGateway-x86_64-${version}.0.msi"
+
+      - name: Upload to OneDrive
+        uses: ./.github/workflows/upload-onedrive
+        with:
+          client_id: ${{ secrets.ONEDRIVE_AUTOMATION_CLIENT_ID }}
+          client_secret: ${{ secrets.ONEDRIVE_AUTOMATION_CLIENT_SECRET }}          
+          remote: releases
+          destination_path: /Gateway/${{ steps.prepare.outputs.version }}
+          conflict_behavior: replace
+          files: ${{ steps.prepare.outputs.files-to-upload }}
```

Comment:
Le fichier ici possède bien les 4 chiffres (2022.2.2.0) ??  J'essaie de suivre, mais je crois que c'est dans le preflight que ça se passe.

Reply:
Yes, the directory and .msi both get a 4-part version (ending in .0)

Reply:
I’ll add a comment at the appropriate place to highlight this!